appinventor/components/src/com/google/appinventor/components/runtime/ActivityStarter.java

```diff
@@ -425,6 +453,15 @@ private Intent buildActivityIntent() {
       intent.putExtra(extraKey, extraValue);
     }
 
+    for (Object extra : extras.toArray()) {
+      YailList castExtra = (YailList) extra;
+      String key = castExtra.getString(0);
```

Comment:
Does getString work on all Yail data types?   If not, you'll need a way to ensure that these elements are strings, and we'll have to think about how to do that.